src/functions.date.js

```diff
@@ -108,24 +108,52 @@ for (let unit in s) {
 	}, {multiValued: true});
 }
 
-_.duration = $.extend(function(ms) {
-	var count = ms || 0;
-	var unit = "ms";
-
-	for (let nextUnit in s) {
-		var nextCount = _.msTo(nextUnit, ms);
+_.duration = $.extend(function (ms, terms) {
+	if (ms === 0 || terms === undefined) {
+		terms = 1;
+	}
 
-		if (nextCount === 0) {
-			break;
+	let timeLeft = ms || 0;
+	let ret = [];
+	let units = Object.keys(s).reverse();
+	units.push("ms");
+	let unitsUsed = [];
+	let onlyConsecutiveNonzeros = true;
+
+	units.map(function(unit) {
+		// get largest value of time unit for the remaining
+		// time to account for
+		let unitValue = unit === "ms"? timeLeft : _.msTo(unit, timeLeft);
+		timeLeft -= unitValue * (unit === "ms" ? 1 : Mavo.Functions[unit]());
+		return unitValue;
+	}).forEach(function (currentValue, index) {
+		if (currentValue !== 0 && ret.length < terms) {
+			let unitProperPlurality = currentValue === 1 && units[index] !== "ms" ? units[index].slice(0, -1) : units[index];
+			ret.push(currentValue + " " + Mavo.Functions.phrase($this, unitProperPlurality));
```

Comment:
Parul, did this code run? `$this` will throw an error here, check out the code you're modifying: it includes `this`, not `$this`. `$this` is only a thing within Mavo expressions.

Reply:
I'm running the duration code in the mavo.es5.js file, which doesn't seem to incorporate the removal of the `$this` param for the function, so I had just assumed for the src file I'd manually take that out. But I forgot to remove it from this phrase function call, thanks for the catch! Is there a way to have the mavo.es5.js file run the same duration version (without `$this`) as the src does?

Reply:
Wait, you should never be manually modifying files in `dist/`, those should be generated by gulp. You only modify files in `src`.
`$this` previously was a function argument. If it's removed as a function argument, it would throw. Which bundle you're running off shouldn't have anything to do with it unless something weird is going on.